src/sas/qtgui/Utilities/TabbedModelEditor.py

```diff
@@ -371,57 +380,41 @@ def checkModel(self, model_str):
             # Set the status bar message
             # GuiUtils.Communicate.statusBarUpdateSignal.emit("Model check failed")
             self.parent.communicate.statusBarUpdateSignal.emit("Model check failed")
-            # Put a thick, red border around the mini-editor
-            self.tabWidget.currentWidget().txtEditor.setStyleSheet("border: 5px solid red")
-            # last_lines = traceback.format_exc().split('\n')[-4:]
-            traceback_to_show = '\n'.join(last_lines)
-            self.tabWidget.currentWidget().txtEditor.setToolTip(traceback_to_show)
+
+            # Put a thick, red border around the editor.
+            from sas.qtgui.Utilities.CodeEditor import QCodeEditor
+
+            # Find all QTextBrowser and QCodeEditor children
+            text_browsers = self.tabWidget.currentWidget().findChildren(QtWidgets.QTextBrowser)
+            code_editors = self.tabWidget.currentWidget().findChildren(QCodeEditor)
+
+            # Combine the lists and apply the stylesheet
+            for child in text_browsers + code_editors:
+                child.setStyleSheet("border: 5px solid red")
+                # last_lines = traceback.format_exc().split('\n')[-4:]
+                traceback_to_show = '\n'.join(last_lines)
+                child.setToolTip(traceback_to_show)
+
             # attempt to find the failing command line number, usually the last line with
             # `File ... line` syntax
-            for line in reversed(all_lines):
-                if 'File' in line and 'line' in line:
+            reversed_error_text = list(reversed(all_lines))
+            for line in reversed_error_text:
```

Comment:
Better to use iterator form `for line in reversed(all_lines)` rather than turning it into a list that you are just going to throw away.

If you do need to assign an iterator to a variable, wrapping it in a tuple is a good defensive practice. Too often I've tried to reuse a sequence only to find that it empty the second time I traverse it. Tuples are immutable and slightly faster.